Fix character starting as overweight

matshou opened the pull request:
<!-- HOW TO USE: Under each "#### Heading" below, enter information relevant to your pull request.
Leave the headings unless they don't apply to your PR.

Please read carefully and don't delete the comments delimited by "< !--" and "-- >"
Once a pull request is submitted automatic stylistic and consistency checks will be performed on the PR's changes.
The results of these can be either seen under the "Files changed" section of a PR or in the check's details.

NOTE: Please grant permission for repository maintainers to edit your PR.  It is EXTREMELY common for PRs to be held up due to trivial changes being requested and the author being unavailable to make them. -->

#### Summary
Bugfixes "Fixed character starting as overweight"
<!-- This section should consist of exactly one line, edit the one above.
Category must be one of these: Features, Content, Interface, Mods, Balance, Bugfixes, Performance, Infrastructure, Build, I18N. Or replace the whole line with just the word None for no changelog entry.
For more on the meaning of each category, see:
https://github.com/CleverRaven/Cataclysm-DDA/blob/master/doc/CHANGELOG_GUIDELINES.md
If approved and merged, your summary will be added to the project changelog:
https://github.com/CleverRaven/Cataclysm-DDA/blob/master/data/changelog.txt -->

#### Purpose of change

Fixes #51366

#### Describe the solution

I've changed the starting calorie value for character gut and stomach to `0` which normalizes starting weight.

#### Testing

I've started a new game with a survivor in evacuee scenario and confirmed that starting calorie value is `54993`.

#### Additional context

![cataclysm-tiles_D8JLzQ84n3](https://user-images.githubusercontent.com/16436920/148684233-be491b14-ca23-4154-be58-9c9f4e4e570d.png)

ZhilkinSerg commented:
You need to update tests with that change.

Also need to verify whether it is still somehow possible to start overweight.

matshou commented:
> You need to update tests with that change.

I assume you mean unit tests. I haven't learned yet how to deal with tests so I could use some help with that.

> Also need to verify whether it is still somehow possible to start overweight.

Do you have any ideas for further testing?

nexusmrsep commented:
Why not reducing stored calories instead? Stomach and gut contents are there to delay initial hunger I think.

Maleclypse commented:
> Why not reducing stored calories instead? Stomach and gut contents are there to delay initial hunger I think.

I came here to second this.  I'm pretty certain this would revert the fix for the bug that players always start hungry.

NetSysFire commented:
Not sure about this either. If you want to just get rid of the "Overweight" in the sidebar somehow it might be fine, but the tiny buffer of more calories is still something useful that lowers the difficulty at the start a tiny bit.

matshou commented:
> Why not reducing stored calories instead? Stomach and gut contents are there to delay initial hunger I think.

I agree that this is a better solution and I've pushed the change.

matshou commented:
> Not sure about this either. If you want to just get rid of the "Overweight" in the sidebar somehow it might be fine, but the tiny buffer of more calories is still something useful that lowers the difficulty at the start a tiny bit.

Lowering the starting calorie amount by 1.4% shouldn't affect the difficulty in any shape of form.
It should still take the character well over a month of not eating to starve.

matshou commented:
> > Why not reducing stored calories instead? Stomach and gut contents are there to delay initial hunger I think.
> 
> I came here to second this. I'm pretty certain this would revert the fix for the bug that players always start hungry.

I've reverted the changes that removed calories from stomach so the player will not start off hungry. The only thing to think about is how much calories should be removed from the starting amount. I've removed `1000` calories as a safety buffer to make sure the digested calories don't bump stored calories over `55000`. The removed value could be lowered down to '500` calories if we are really concerned about the impact on difficulty but that would easily make the player overweight after his first meal.

kevingranade commented:
1,000 Calories is fine, that's just one meal.
